src/controllers/product.controller.ts

```diff
@@ -0,0 +1,27 @@
+import { Request, Response } from 'express';
```

Comment:
application is not scalable, so one of the possible solution is to convert this file into  *.roter.ts

```

import express, { Request, Response } from "express";
import { ProductService } from "../services/product.service";
const router = express.Router();
export default router;

router.get('/', (req: Request, res: Response) => {
   try {
    console.log('1');
    const products = ProductService.getAllProducts();
    res.status(200).json({ data: products, error: null });
  } catch (error) {
    res.status(500).json({ data: null, error: "error.message" });
  }
});

router.get("/:productId", (req: Request, res: Response) => {
  try {
    const productId = req.params.productId;
    const product = ProductService.getProductById(productId);
    if (!product) {
      res.status(404).json({ data: null, error: "No product with such id" });
      return;
    }
    res.status(200).json({ data: product, error: null });
  } catch (error) {
    res.status(500).json({ data: null, error: "error.message" });
  }
});

```